In [2]:
import pandas as pd
import numpy as  np

<h1><center> QUIZ </center></h1>

Este Quiz tiene el objetivo de, además de utilizar Python de manera práctica, **generar conciencia y presentar herramientas que puedan ser utilizadas por los estudiandtes para analizar de forma crítica y reflexiva los problemas de nuestras sociedad**

Para tal fin, el quiz toma como recurso de análisis el Censo Nacional de Población y Vivienda 2018

> http://microdatos.dane.gov.co/index.php/catalog/643/data_dictionary

allí podrán encontrar las bases de datos, la descripción de las variables y la metodología utilizada

<h1><center> Indicadores de Pobreza y Desigualdad en el Departamento del Vaupés </center></h1>

Use las bases de datos:

    VIVIENDAS: CNPV2018_1VIV_A2_97.CSV
    HOGARES: CNPV2018_2HOG_A2_97.CSV
    PERSONAS: CNPV2018_5PER_A2_97.CSV

In [4]:
viviendas = pd.read_csv('D:/maestria_ciencias_economicas_unal/curso_python/base_dane_quiz/97Vaupes/CNPV2018_1VIV_A2_97.CSV')
hogares = pd.read_csv('D:/maestria_ciencias_economicas_unal/curso_python/base_dane_quiz/97Vaupes/CNPV2018_2HOG_A2_97.CSV')
personas = pd.read_csv('D:/maestria_ciencias_economicas_unal/curso_python/base_dane_quiz/97Vaupes/CNPV2018_5PER_A2_97.CSV')

In [20]:
viviendas.dtypes

TIPO_REG               int64
U_DPTO                 int64
U_MPIO                 int64
UA_CLASE               int64
U_EDIFICA              int64
COD_ENCUESTAS          int64
U_VIVIENDA             int64
UVA_ESTATER          float64
UVA1_TIPOTER         float64
UVA2_CODTER          float64
UVA_ESTA_AREAPROT      int64
UVA1_COD_AREAPROT    float64
UVA_USO_UNIDAD         int64
V_TIPO_VIV           float64
V_CON_OCUP           float64
V_TOT_HOG            float64
V_MAT_PARED          float64
V_MAT_PISO           float64
VA_EE                float64
VA1_ESTRATO          float64
VB_ACU               float64
VC_ALC               float64
VD_GAS               float64
VE_RECBAS            float64
VE1_QSEM             float64
VF_INTERNET          float64
V_TIPO_SERSA         float64
L_TIPO_INST          float64
L_EXISTEHOG          float64
L_TOT_PERL           float64
dtype: object

In [21]:
viviendas.shape

(7471, 30)

In [27]:
viviendas.columns.to_list()

['TIPO_REG',
 'U_DPTO',
 'U_MPIO',
 'UA_CLASE',
 'U_EDIFICA',
 'COD_ENCUESTAS',
 'U_VIVIENDA',
 'UVA_ESTATER',
 'UVA1_TIPOTER',
 'UVA2_CODTER',
 'UVA_ESTA_AREAPROT',
 'UVA1_COD_AREAPROT',
 'UVA_USO_UNIDAD',
 'V_TIPO_VIV',
 'V_CON_OCUP',
 'V_TOT_HOG',
 'V_MAT_PARED',
 'V_MAT_PISO',
 'VA_EE',
 'VA1_ESTRATO',
 'VB_ACU',
 'VC_ALC',
 'VD_GAS',
 'VE_RECBAS',
 'VE1_QSEM',
 'VF_INTERNET',
 'V_TIPO_SERSA',
 'L_TIPO_INST',
 'L_EXISTEHOG',
 'L_TOT_PERL']

# 1 VIVIENDAS

Use las columnas:
    
    VIVIENDAS: ["TIPO_REG","U_DPTO","U_MPIO","COD_ENCUESTAS","UVA_ESTATER","V_MAT_PARED","V_MAT_PISO","VB_ACU",  "V_TIPO_SERSA" ]
    
                UVA_ESTATER ----> Vivienda en una territorialidad étnica
                                    1. Si
                                    2. No
                V_MAT_PARED ----> Material predominante en paredes exteriores
                                    1 Bloque, ladrillo, piedra, madera pulida
                                    2 Concreto vaciado
                                    3 Material prefabricado
                                    4 Guadua
                                    5 Tapia pisada, bahareque, adobe
                                    6 Madera burda, tabla, tablon
                                    7 Caña, esterilla, otros vegetales
                                    8 Materiales de deshecho (Zinc, tela, carton, latas, plasticos, otros)
                                    9 No tiene paredes
                V_MAT_PISO ----> Material predominante en los pisos
                                    1 Marmol, parque, madera pulida y lacada
                                    2 Baldosa, vinilo, tableta, ladrillo, laminado
                                    3 Alfombra
                                    4 Cemento, gravilla
                                    5 Madera burda, tabla, tablon, otro vegetal
                                    6 Tierra, arena, barro
                VB_ACU    ----> Cuenta con servicio de acueducto
                                    1 Si
                                    2 No
                V_TIPO_SERSA --> Tipo de servicio sanitario (inodoro)
                                    1 Inodoro conectado al alcantarillado?
                                    2 Inodoro conectado a pozo septico?
                                    3 Inodoro sin conexion?
                                    4 Letrina?
                                    5 Inodoro con descarga directa a fuentes de agua (bajamar)?
                                    6 esta vivienda No tiene servicio sanitario?

In [28]:
viviendas = viviendas[["TIPO_REG","U_DPTO","U_MPIO","COD_ENCUESTAS","UVA_ESTATER","V_MAT_PARED","V_MAT_PISO","VB_ACU","V_TIPO_SERSA" ]]

### Preguntas

 1. ¿Cuantas viviendas hay con territorialidad étnica?
 2. ¿Cuál es el material predominante en paredes exteriores?
 3. ¿Cuál es el material predominante en los pisos?
 4. ¿Cuántas viviendas NO cuentan con servicio de alcantarillado?
 5. ¿Cuál es el tipo de servicio sanitario (inodoro) predominante?

In [44]:
viviendas["UVA_ESTATER"].value_counts() #4210

1.0    4210
2.0     353
Name: UVA_ESTATER, dtype: int64

In [46]:
viviendas["V_MAT_PARED"].value_counts().sort_values() #madera burda, tabla, tablon

3.0      26
4.0      55
5.0      62
9.0      65
8.0     185
2.0     204
7.0     810
1.0     961
6.0    4461
Name: V_MAT_PARED, dtype: int64

In [47]:
viviendas["V_MAT_PISO"].value_counts().sort_values() #madera burda, tabla, tablon, otro vegetal

3.0       2
1.0      73
2.0     439
4.0    1211
6.0    2320
5.0    2784
Name: V_MAT_PISO, dtype: int64

In [49]:
viviendas["VB_ACU"].value_counts().sort_values() #4436

1.0    2393
2.0    4436
Name: VB_ACU, dtype: int64

In [50]:
viviendas["V_TIPO_SERSA"].value_counts().sort_values() # esta vivienda No tiene servicio sanitario?

4.0      34
5.0      37
3.0      57
9.0     247
2.0    1032
1.0    1974
6.0    3448
Name: V_TIPO_SERSA, dtype: int64

# 2. HOGARES

use las columnas:

    HOGARES: ["TIPO_REG","U_DPTO","U_MPIO","COD_ENCUESTAS","H_NRO_CUARTOS","H_DONDE_PREPALIM","H_AGUA_COCIN","HA_TOT_PER"]
    
             H_NRO_CUARTOS     ---> Número de cuartos en total
             
             H_DONDE_PREPALIM  ---> Lugar donde preparan los alimentos
                                    1 En un cuarto usado solo para cocinar?
                                    2 En un cuarto usado también para dormir?
                                    3 En una sala-comedor con lavaplatos?
                                    4 En una sala-comedor sin lavaplatos?
                                    5 En un patio, corredor, enramada o al aire libre?
                                    6 No preparan alimentos en la vivienda?
                                    9 No informa
            
             H_AGUA_COCIN    ----> Fuente de agua para preparar los alimentos
                                    1 Acueducto público?
                                    2 Acueducto veredal?
                                    3 Red de distribución comunitaria?
                                    4 Pozo con bomba?
                                    5 Pozo sin bomba, aljibe, jaguey o barreno?
                                    6 Agua lluvia?
                                    7 RÍo, quebrada, manantial, nacimiento?
                                    8 Pila pública?
                                    9 Carrotanque?
                                    10 Aguatero?
                                    11 Agua embotellada o en bolsa?
                                    12 no preparan alimentos
                                    99 No informa
                                    
             HA_TOT_PER      ----> Total personas en el hogar                  

In [51]:
hogares = hogares[["TIPO_REG","U_DPTO","U_MPIO","COD_ENCUESTAS","H_NRO_CUARTOS","H_DONDE_PREPALIM","H_AGUA_COCIN","HA_TOT_PER"]]

### Preguntas
 1. ¿Cuál es el número de cuartos predominantes?
 2. ¿Cuál es el lugar más usual para preparas los alimentos?
 3. ¿Cuál es la fuente de agua más usual para preparar los alimentos?
 4. ¿Cual es el número predominante de personas que vien en el hogar?
 5. ¿En promedio cuantas personas viven en el hogar?

In [53]:
hogares["H_NRO_CUARTOS"].value_counts().sort_values(ascending = False) # 2 cuartos

2.0     1768
1.0     1761
3.0     1497
4.0      941
5.0      542
99.0     200
6.0      198
7.0       57
8.0       20
9.0       15
10.0       9
12.0       4
15.0       3
11.0       2
13.0       2
14.0       1
Name: H_NRO_CUARTOS, dtype: int64

In [54]:
hogares["H_DONDE_PREPALIM"].value_counts().sort_values(ascending = False) #En un cuarto usado solo para cocinar

1.0    5332
2.0     650
3.0     255
9.0     254
6.0     195
4.0     182
5.0     152
Name: H_DONDE_PREPALIM, dtype: int64

In [55]:
hogares["H_AGUA_COCIN"].value_counts().sort_values(ascending = False) #Agua lluvia

6.0     4168
7.0     1159
11.0     690
1.0      386
99.0     251
2.0       51
4.0       47
5.0       42
3.0       15
10.0       9
8.0        7
Name: H_AGUA_COCIN, dtype: int64

In [56]:
hogares["HA_TOT_PER"].value_counts().sort_values(ascending = False) # una persona por hogar

1.0     1132
3.0     1109
4.0     1052
2.0      903
5.0      872
6.0      708
7.0      690
8.0      249
9.0      152
10.0      70
11.0      47
12.0      14
14.0      12
13.0       6
15.0       3
16.0       1
Name: HA_TOT_PER, dtype: int64

In [57]:
hogares["HA_TOT_PER"].mean() #4.125 personas por hogar

4.125783475783476

# 3. PERSONAS

Use las columnas:
                   
        PERSONAS: ["TIPO_REG","U_DPTO","U_MPIO","COD_ENCUESTAS","PA1_GRP_ETNIC", "PA11_COD_ETNIA", "P_QUEHIZO_PPAL",
        "P_ALFABETA", "P_NIVEL_ANOSR"]
        
        
        
                PA1_GRP_ETNIC ---- > Reconocimiento étnico
                                        1 Indígena?
                                        2 Gitano(a) o Rrom?
                                        3 Raizal del Archipielago de San Andrés, Providencia y Santa Catalina?
                                        4 Palenquero(a) de San Basilio?
                                        5 Negro(a), Mulato(a), Afrodescendiente, Afrocolombiano(a)?
                                        6 Ningún grupo étnico
                                        9 No Informa
               PA11_COD_ETNIA --->  Pueblo indigena de pertenencia
               
                                        Achagua	10
                                        Amorua	20
                                        Uipiue	21
                                        Yamalero	25
                                        Yaruro	26
                                        Andoke	30
                                        Arhuaca	40
                                        Arsaria	50
                                        Bara	60
                                        Barasana	70
                                        Bari	80
                                        Betoye	90
                                        Bora	100
                                        Cabiyari	110
                                        Carabayo	120
                                        Carapana	130
                                        Carifuna	140
                                        Chimila	150
                                        Chiricoa	160
                                        Cocama	170
                                        Coconuco	180
                                        Corebahu	190
                                        Coyaima	200
                                        Awa	210
                                        Cobewa	220
                                        Cuiba	230
                                        Bayano	240
                                        Curripaca	250
                                        Baniba	251
                                        Guariquema	252
                                        Desana	260
                                        Dujos	270
                                        Ebera	280
                                        Catio	281
                                        Chami	282
                                        Eepera	283
                                        Embera-Dobida	284
                                        Nutabe	285
                                        Guambiana	290
                                        Ambalo	291
                                        Kizgo	292
                                        Guanaca	300
                                        Guanana	310
                                        Cunimia	320
                                        Cañamomo	330
                                        Imga	340
                                        Cache	350
                                        Cofan	360
                                        Coghi	370
                                        Detuama	380
                                        Auirubain	390
                                        Hitnu	400
                                        Macaguane	401
                                        Buhagana	410
                                        Makusa	430
                                        Cacua	431
                                        Hupdu	432
                                        Juhup	433
                                        Judpa	434
                                        Macu	435
                                        Maiben	440
                                        Jupichiya	450
                                        Hurumi	451
                                        Je´Eruriwa	455
                                        Miraña	460
                                        Chibcha	470
                                        Nonuya	480
                                        Diocaya	490
                                        Nasa	500
                                        Polindara	501
                                        Andakies	505
                                        Cuipaco	510
                                        Dearua	520
                                        Parata-Puya	530
                                        Papiua	540
                                        Guaipunare	550
                                        Pasto	560
                                        Killacinga	565
                                        Saliba	570
                                        Guahiba	580
                                        Mapayerri	585
                                        Ceona	590
                                        Chiranga	600
                                        Eduria	610
                                        Ohañara	620
                                        Tanigua	621
                                        Tariana	630
                                        Juna	640
                                        Totoro	650
                                        Ticuna	660
                                        Pasee	661
                                        Mariposo	670
                                        Dasea	680
                                        Lache	690
                                        Dojkapuara	700
                                        Noanama	710
                                        Guajiro	720
                                        Bue	730
                                        Muinane	731
                                        Yari	732
                                        Mishara	740
                                        Mitimae	750
                                        Camejeya	760
                                        Mapapi	770
                                        Yuco	780
                                        Totsoca	790
                                        Senu	800
                                        Guane	810
                                        Mokana	820
                                        Otavaleña	830
                                        Kichwa	840
                                        Kamkuamo	850
                                        Tayronas	855
                                        Chitarero	860
                                        Quimbaya	870
                                        Calima	880
                                        Panches	890
                                        Indigenas-Ecuador-(Diferente-De-Otavaleños)	900
                                        Indigenas-Peru	910
                                        Indigenas-Venezuela	920
                                        Indigenas-Mexico	930
                                        Indigenas-Brasil	940
                                        Ñengantu	941
                                        Indigenas-Panama	950
                                        Indigenas-Bolivia	960
                                        Maya-(Guatemala)	970
                                        Indigena-Sin-Informacion	999

                P_QUEHIZO_PPAL ---> Tratamiento principal del problema de salud
                        1 Acudió a la entidad de seguridad social en salud de la cual es filiado(a)?
                        2 Acudió a un médico particular? (general, especialista, odontólogo, terapeuta u otro)
                        3 Acudió a un boticario, farmacéuta, droguista?
                        4 Asistió a terapias alternativas? (acupuntura, esencias florales, musicoterapias, homeópata, etc.)
                        5 Acudió a una autoridad indígena espiritual?
                        6 Acudió a otro médico de un grupo étnico? (curandero, yerbatero, etc.)
                        7 Usó remedios caseros?
                        8 Se autorrecetó
                        9 No hizo nada
                        99 No Informa
                        No Aplica
                        
                P_ALFABETA ---> Sabe leer y escribir
                                    1 SI
                                    2 NO
                                    9 No Informa
                                    No Aplica
                                    
                P_NIVEL_ANOSR ---> Nivel educativo más alto alcanzado y último año o grado aprobado en ese nivel (recodificado)
                                    1 Preescolar
                                    2 Básica primaria
                                    3 Básica secundaria
                                    4 Media academica o clasica
                                    5 Media tecnica
                                    6 Normalista
                                    7 Técnica profesional o Tecnológica
                                    8 Universitario
                                    9 Especialización, maestría, doctorado
                                    10 Ninguno
                                    99 No Informa
                                    No Aplica

In [59]:
personas = personas[["TIPO_REG","U_DPTO","U_MPIO","COD_ENCUESTAS","PA1_GRP_ETNIC", "PA11_COD_ETNIA", "P_QUEHIZO_PPAL",
    "P_ALFABETA", "P_NIVEL_ANOSR"]]

### Preguntas

 1. ¿Cuál es el Reconocimiento étnico predominante?
 2. ¿Cuál es el pueblo indígena de pertenencia predominante?
 3. ¿Cuál es el tratamiento principal del problema de salud predominante?
 4. ¿Cuántas personas NO saben escribir? 
 5. ¿Cuántas personas estudiaron en la universidad? 

In [61]:
personas["PA1_GRP_ETNIC"].value_counts().sort_values(ascending = False) #indigena

1    30787
6     5433
9     1182
5      270
3       10
4        8
Name: PA1_GRP_ETNIC, dtype: int64

In [62]:
personas["PA11_COD_ETNIA"].value_counts().sort_values(ascending = False) #Cobewa, Desana 

220.0    11457
260.0     2879
310.0     2689
680.0     2640
410.0     1423
600.0     1421
700.0     1332
640.0      887
130.0      860
790.0      782
60.0       763
530.0      644
70.0       639
250.0      626
110.0      397
620.0      322
630.0      173
770.0      171
430.0      162
540.0      144
610.0       82
550.0       47
999.0       31
730.0       30
380.0       30
660.0       17
140.0       16
251.0       14
570.0       12
320.0       12
431.0       11
731.0       11
510.0        9
760.0        8
460.0        8
450.0        7
720.0        4
580.0        4
500.0        3
100.0        3
490.0        3
350.0        2
840.0        2
560.0        1
20.0         1
180.0        1
750.0        1
940.0        1
390.0        1
290.0        1
470.0        1
282.0        1
280.0        1
Name: PA11_COD_ETNIA, dtype: int64

In [63]:
personas["P_QUEHIZO_PPAL"].value_counts().sort_values(ascending = False) #Acudió a la entidad de seguridad social en salud de la cual es filiado

1.0    1036
0.0     252
5.0     209
6.0     176
7.0     146
9.0     141
3.0      93
8.0      90
2.0      77
4.0       5
Name: P_QUEHIZO_PPAL, dtype: int64

In [64]:
personas["P_ALFABETA"].value_counts().sort_values(ascending = False) # 3178 personas no saben leer ni escribir

1.0    26404
9.0     4928
2.0     3178
Name: P_ALFABETA, dtype: int64

In [65]:
personas["P_NIVEL_ANOSR"].value_counts().sort_values(ascending = False) # 858 personas estudiaron en la universidad

2.0     15422
3.0      7687
4.0      3736
10.0     1848
99.0     1668
1.0      1283
7.0       933
8.0       858
6.0       379
5.0       360
9.0       336
Name: P_NIVEL_ANOSR, dtype: int64

# 4 Análisis conjunto

 1. Use las tres bases de datos ``viviendas``, ``hogares`` y ``personas``, con las columnas especificadas en cada punto (1,2,3) 
 2. tome la columna "COD_ENCUESTAS" y junte las tres bases de datos con pd.merge()

In [68]:
viviendas.shape

(7471, 9)

In [87]:
viviendas["COD_ENCUESTAS"].duplicated().sum()

0

In [71]:
viviendas.head(5)

TIPO_REG  U_DPTO  U_MPIO  COD_ENCUESTAS  UVA_ESTATER  V_MAT_PARED  \
0         1      97       1        7888115          NaN          1.0   
1         1      97       1        7888128          NaN          NaN   
2         1      97       1        7888136          NaN          6.0   
3         1      97       1        7888140          NaN          6.0   
4         1      97       1        7888261          NaN          NaN   

   V_MAT_PISO  VB_ACU  V_TIPO_SERSA  
0         4.0     1.0           1.0  
1         NaN     NaN           NaN  
2         6.0     2.0           6.0  
3         6.0     2.0           3.0  
4         NaN     NaN           NaN

In [69]:
hogares.shape

(7151, 8)

In [158]:
hogares.duplicated().sum()

23

In [163]:
hogares["mask_hogar"] = hogares.duplicated()



<ipython-input-163-5a864f76c2d3>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  hogares["mask_hogar"] = hogares.duplicated()


In [173]:
hogares.groupby("mask_hogar").get_group(1) #observar valores duplicados

TIPO_REG  U_DPTO  U_MPIO  COD_ENCUESTAS  H_NRO_CUARTOS  \
30           2      97       1        7888140            4.0   
276          2      97       1       10239075            2.0   
597          2      97       1       13288186            5.0   
623          2      97       1       13288232            3.0   
989          2      97       1       13827332            4.0   
2586         2      97       1        8365700            1.0   
2612         2      97       1        8750594            2.0   
2630         2      97       1        8750844            2.0   
2746         2      97       1        8754924            3.0   
2748         2      97       1        8754924            3.0   
2959         2      97       1        9723793            5.0   
3059         2      97       1       10470423            2.0   
5460         2      97     161       13064995            1.0   
5463         2      97     161       13064995            1.0   
5464         2      97     161       13064995            1.0   
5465         2      97     161       13064995            1.0   
5469         2      97     161       13065001            5.0   
5522         2      97     161       13065143            3.0   
5540         2      97     161       13065219            3.0   
6554         2      97     666       15524803            1.0   
6561         2      97     666       15524840            1.0   

      H_DONDE_PREPALIM  H_AGUA_COCIN  HA_TOT_PER  mask_hogar  
30                 1.0           6.0         2.0        True  
276                1.0           6.0         3.0        True  
597                1.0           6.0         3.0        True  
623                1.0           6.0         4.0        True  
989                1.0           6.0         3.0        True  
2586               1.0           6.0         3.0        True  
2612               1.0           6.0         2.0        True  
2630               1.0           6.0         1.0        True  
2746               1.0           6.0         1.0        True  
2748               1.0           6.0         1.0        True  
2959               1.0           6.0         2.0        True  
3059               1.0           6.0         3.0        True  
5460               6.0           NaN         1.0        True  
5463               6.0           NaN         1.0        True  
5464               6.0           NaN         1.0        True  
5465               6.0           NaN         1.0        True  
5469               1.0           6.0         4.0        True  
5522               1.0           6.0         2.0        True  
5540               1.0           6.0         1.0        True  
6554               6.0           NaN         1.0        True  
6561               6.0           NaN         2.0        True

In [70]:
personas.shape

(37690, 9)

In [73]:
personas.head(5)

TIPO_REG  U_DPTO  U_MPIO  COD_ENCUESTAS  PA1_GRP_ETNIC  PA11_COD_ETNIA  \
0         5      97       1        7888026              1           220.0   
1         5      97       1        7888026              1           220.0   
2         5      97       1        7888026              1           220.0   
3         5      97       1        7888026              1           220.0   
4         5      97       1        7888026              1           220.0   

   P_QUEHIZO_PPAL  P_ALFABETA  P_NIVEL_ANOSR  
0             NaN         1.0            4.0  
1             NaN         1.0            2.0  
2             NaN         1.0            1.0  
3             NaN         1.0            4.0  
4             NaN         1.0            2.0

In [142]:
personas["COD_ENCUESTAS"].duplicated().sum() #pregunta


30730

In [80]:
personas["COD_ENCUESTAS"].value_counts()

750000613    571
750000754    534
750000785    491
750000894    319
750000638    310
            ... 
902427339      1
900823706      1
600011355      1
900336679      1
900736865      1
Name: COD_ENCUESTAS, Length: 6960, dtype: int64

In [145]:
data1 = pd.merge(left= viviendas, right= hogares, on = 'COD_ENCUESTAS', how = 'inner')
data1.head(10) 

TIPO_REG_x  U_DPTO_x  U_MPIO_x  COD_ENCUESTAS  UVA_ESTATER  V_MAT_PARED  \
0           1        97         1        7888115          NaN          1.0   
1           1        97         1        7888136          NaN          6.0   
2           1        97         1        7888140          NaN          6.0   
3           1        97         1        7888140          NaN          6.0   
4           1        97         1        7888262          NaN          4.0   
5           1        97         1        7888263          NaN          4.0   
6           1        97         1        7888266          NaN          2.0   
7           1        97         1        7888285          NaN          1.0   
8           1        97         1        7888307          NaN          6.0   
9           1        97         1        7888308          NaN          6.0   

   V_MAT_PISO  VB_ACU  V_TIPO_SERSA  TIPO_REG_y  U_DPTO_y  U_MPIO_y  \
0         4.0     1.0           1.0           2        97         1   
1         6.0     2.0           6.0           2        97         1   
2         6.0     2.0           3.0           2        97         1   
3         6.0     2.0           3.0           2        97         1   
4         6.0     2.0           6.0           2        97         1   
5         6.0     2.0           6.0           2        97         1   
6         4.0     1.0           1.0           2        97         1   
7         4.0     1.0           1.0           2        97         1   
8         5.0     2.0           6.0           2        97         1   
9         5.0     2.0           2.0           2        97         1   

   H_NRO_CUARTOS  H_DONDE_PREPALIM  H_AGUA_COCIN  HA_TOT_PER  
0            5.0               1.0          11.0         5.0  
1            1.0               1.0           6.0         1.0  
2            4.0               1.0           6.0         2.0  
3            4.0               1.0           6.0         2.0  
4            4.0               1.0           6.0         1.0  
5            5.0               1.0           6.0         1.0  
6            5.0               1.0          11.0         2.0  
7            5.0               1.0          11.0         2.0  
8            1.0               2.0           6.0         2.0  
9            4.0               1.0           4.0         2.0

In [146]:
data1.duplicated().sum()

23

In [147]:
data1["mask"]= data1.duplicated()
  #Datos duplicados luego de realizar el merge, quiero ver las filas, cuales son


In [149]:
data1.head()

TIPO_REG_x  U_DPTO_x  U_MPIO_x  COD_ENCUESTAS  UVA_ESTATER  V_MAT_PARED  \
0           1        97         1        7888115          NaN          1.0   
1           1        97         1        7888136          NaN          6.0   
2           1        97         1        7888140          NaN          6.0   
3           1        97         1        7888140          NaN          6.0   
4           1        97         1        7888262          NaN          4.0   

   V_MAT_PISO  VB_ACU  V_TIPO_SERSA  TIPO_REG_y  U_DPTO_y  U_MPIO_y  \
0         4.0     1.0           1.0           2        97         1   
1         6.0     2.0           6.0           2        97         1   
2         6.0     2.0           3.0           2        97         1   
3         6.0     2.0           3.0           2        97         1   
4         6.0     2.0           6.0           2        97         1   

   H_NRO_CUARTOS  H_DONDE_PREPALIM  H_AGUA_COCIN  HA_TOT_PER   mask  
0            5.0               1.0          11.0         5.0  False  
1            1.0               1.0           6.0         1.0  False  
2            4.0               1.0           6.0         2.0  False  
3            4.0               1.0           6.0         2.0   True  
4            4.0               1.0           6.0         1.0  False

In [157]:
data_duplicados = data1.groupby("mask").get_group(1) #por que se repiten
data_duplicados

TIPO_REG_x  U_DPTO_x  U_MPIO_x  COD_ENCUESTAS  UVA_ESTATER  V_MAT_PARED  \
3              1        97         1        7888140          NaN          6.0   
526            1        97         1       13288186          NaN          1.0   
1234           1        97         1       10239075          NaN          6.0   
1394           1        97         1       13827332          NaN          1.0   
1867           1        97         1       13288232          NaN          6.0   
2509           1        97         1        8365700          1.0          7.0   
2847           1        97         1        8754924          1.0          6.0   
2848           1        97         1        8754924          1.0          6.0   
2849           1        97         1        8754924          1.0          6.0   
3095           1        97         1        8750844          1.0          6.0   
3322           1        97         1        9723793          1.0          6.0   
3330           1        97         1       10470423          1.0          6.0   
5127           1        97         1        8750594          1.0          6.0   
5416           1        97       161       13064995          NaN          6.0   
5418           1        97       161       13064995          NaN          6.0   
5419           1        97       161       13064995          NaN          6.0   
5420           1        97       161       13064995          NaN          6.0   
5421           1        97       161       13064995          NaN          6.0   
5424           1        97       161       13065001          NaN          6.0   
5434           1        97       161       13065219          NaN          6.0   
5569           1        97       161       13065143          NaN          6.0   
6540           1        97       666       15524803          NaN          6.0   
6663           1        97       666       15524840          NaN          6.0   

      V_MAT_PISO  VB_ACU  V_TIPO_SERSA  TIPO_REG_y  U_DPTO_y  U_MPIO_y  \
3            6.0     2.0           3.0           2        97         1   
526          4.0     1.0           1.0           2        97         1   
1234         6.0     1.0           1.0           2        97         1   
1394         4.0     1.0           1.0           2        97         1   
1867         6.0     1.0           2.0           2        97         1   
2509         6.0     2.0           6.0           2        97         1   
2847         5.0     2.0           6.0           2        97         1   
2848         5.0     2.0           6.0           2        97         1   
2849         5.0     2.0           6.0           2        97         1   
3095         5.0     2.0           6.0           2        97         1   
3322         5.0     2.0           6.0           2        97         1   
3330         6.0     2.0           2.0           2        97         1   
5127         5.0     2.0           6.0           2        97         1   
5416         5.0     1.0           1.0           2        97       161   
5418         5.0     1.0           1.0           2        97       161   
5419         5.0     1.0           1.0           2        97       161   
5420         5.0     1.0           1.0           2        97       161   
5421         5.0     1.0           1.0           2        97       161   
5424         5.0     1.0           1.0           2        97       161   
5434         5.0     2.0           2.0           2        97       161   
5569         5.0     1.0           1.0           2        97       161   
6540         5.0     1.0           6.0           2        97       666   
6663         4.0     1.0           5.0           2        97       666   

      H_NRO_CUARTOS  H_DONDE_PREPALIM  H_AGUA_COCIN  HA_TOT_PER  mask  
3               4.0               1.0           6.0         2.0  True  
526             5.0               1.0           6.0         3.0  True  
1234            2.0               1.0           6.0         3.0  Tru

In [156]:
data_duplicados["TIPO_REG_x"].unique()

array([1], dtype=int64)

In [ ]:
data1.drop_duplicates(keep = 'first',inplace=True) # eliminar datos duplicados
data1

In [174]:
data2 = pd.merge(left=data1,right= personas,on = 'COD_ENCUESTAS', how = 'inner') #Datos duplicados luego de realizar el merge 

In [175]:
data2.duplicated().sum() 

14457

In [176]:
data2

TIPO_REG_x  U_DPTO_x  U_MPIO_x  COD_ENCUESTAS  UVA_ESTATER  \
0               1        97         1        7888115          NaN   
1               1        97         1        7888115          NaN   
2               1        97         1        7888115          NaN   
3               1        97         1        7888115          NaN   
4               1        97         1        7888115          NaN   
...           ...       ...       ...            ...          ...   
39199           1        97       889      750001214          1.0   
39200           1        97       889      750001214          1.0   
39201           1        97       889      750001214          1.0   
39202           1        97       889      750001214          1.0   
39203           1        97       889      750001214          1.0   

       V_MAT_PARED  V_MAT_PISO  VB_ACU  V_TIPO_SERSA  TIPO_REG_y  ...  \
0              1.0         4.0     1.0           1.0           2  ...   
1              1.0         4.0     1.0           1.0           2  ...   
2              1.0         4.0     1.0           1.0           2  ...   
3              1.0         4.0     1.0           1.0           2  ...   
4              1.0         4.0     1.0           1.0           2  ...   
...            ...         ...     ...           ...         ...  ...   
39199          NaN         NaN     NaN           NaN           2  ...   
39200          NaN         NaN     NaN           NaN           2  ...   
39201          NaN         NaN     NaN           NaN           2  ...   
39202          NaN         NaN     NaN           NaN           2  ...   
39203          NaN         NaN     NaN           NaN           2  ...   

       HA_TOT_PER   mask  TIPO_REG  U_DPTO  U_MPIO  PA1_GRP_ETNIC  \
0             5.0  False         5      97       1              4   
1             5.0  False         5      97       1              4   
2             5.0  False         5      97       1              4   
3             5.0  False         5      97       1              4   
4             5.0  False         5      97       1              4   
...           ...    ...       ...     ...     ...            ...   
39199         NaN  False         5      97     889              1   
39200         NaN  False         5      97     889              1   
39201         NaN  False         5      97     889              1   
39202         NaN  False         5      97     889              1   
39203         NaN  False         5      97     889              1   

       PA11_COD_ETNIA  P_QUEHIZO_PPAL  P_ALFABETA  P_NIVEL_ANOSR  
0                 NaN             NaN         1.0            4.0  
1                 NaN             NaN         1.0            2.0  
2                 NaN             NaN         1.0            2.0  
3                 NaN             NaN         NaN            NaN  
4                 NaN             NaN         1.0            4.0  
...               ...             ...         ...            ...  
39199           310.0             NaN         9.0            2.0  
39200           530.0             NaN         9.0            2.0  
39201           310.0             NaN         9.0            2.0  
39202           260.0             NaN         9.0            2.0  
39203           530.0             NaN         9.0            2.0  

[39204 rows x 25 columns]

In [ ]:
data2.drop_duplicates(keep='first',inplace=True) # eliminar datos duplicados
data2

### Preguntas

Agrupe la base de datos por la columna PA1_GRP_ETNIC ---- > Reconocimiento étnico, tome el grupo de personas que se autoreconoce como indígena y responda las siguientes preguntas:

 1. ¿Un mayor nivel educativo hace que los materiales de la pared sean mejores? haga un scatter plot o gráfico de dispersión, use pivot_table()
 
     pivot_table(index='P_NIVEL_ANOSR', values= ['V_MAT_PARED'], aggfunc=np.mean)
     
     
 2. ¿Las personas que estudiaron una maestría, en promedio tienen una mejor fuente de agua para cocinas, y mejores materiales del piso y la pared?
 
     use groupby

In [178]:
data3 = data2.groupby("PA1_GRP_ETNIC").get_group(1)

In [182]:
pd.pivot_table(data3, index='P_NIVEL_ANOSR', values= ['V_MAT_PARED'], aggfunc=np.mean) #verdadero

V_MAT_PARED
P_NIVEL_ANOSR             
1.0               6.054393
2.0               6.051584
3.0               5.949872
4.0               5.561593
5.0               5.484746
6.0               5.670300
7.0               5.058140
8.0               4.250564
9.0               3.562500
10.0              6.198657
99.0              6.218452

In [185]:
data3.groupby("P_NIVEL_ANOSR").aggregate({"H_AGUA_COCIN":np.mean, "V_MAT_PISO":np.mean, "V_MAT_PARED":np.mean}) #falso

H_AGUA_COCIN  V_MAT_PISO  V_MAT_PARED
P_NIVEL_ANOSR                                       
1.0                9.404795    5.344491     6.054393
2.0                8.030949    5.349751     6.051584
3.0                7.523212    5.252418     5.949872
4.0                7.235449    5.008560     5.561593
5.0                6.872414    5.132203     5.484746
6.0                8.200000    5.029973     5.670300
7.0                7.091043    4.742733     5.058140
8.0                6.645833    4.379233     4.250564
9.0                8.081081    4.178571     3.562500
10.0               9.025381    5.482373     6.198657
99.0              16.100213    5.431601     6.218452

In [188]:
data_4 = data2.copy()

In [190]:
data_4.drop_duplicates(keep='first',inplace=True) # eliminar datos duplicados

In [192]:
data_5 = data_4.groupby("PA1_GRP_ETNIC").get_group(1) #verdadero

In [193]:
pd.pivot_table(data_5, index='P_NIVEL_ANOSR', values= ['V_MAT_PARED'], aggfunc=np.mean)

V_MAT_PARED
P_NIVEL_ANOSR             
1.0               6.050215
2.0               5.993985
3.0               5.930346
4.0               5.546598
5.0               5.440299
6.0               5.648485
7.0               5.092715
8.0               4.208877
9.0               3.626168
10.0              6.150454
99.0              6.216568

In [194]:
data_5.groupby("P_NIVEL_ANOSR").aggregate({"H_AGUA_COCIN":np.mean, "V_MAT_PISO":np.mean, "V_MAT_PARED":np.mean}) #falso

H_AGUA_COCIN  V_MAT_PISO  V_MAT_PARED
P_NIVEL_ANOSR                                       
1.0                9.492754    5.341463     6.050215
2.0                8.103057    5.313282     5.993985
3.0                7.644766    5.260799     5.930346
4.0                7.289548    5.016775     5.546598
5.0                6.908745    5.078358     5.440299
6.0                8.344512    5.048485     5.648485
7.0                7.169179    4.766556     5.092715
8.0                6.669355    4.362924     4.208877
9.0                8.179245    4.196262     3.626168
10.0               8.423429    5.461089     6.150454
99.0              15.886905    5.424852     6.216568